### ISSUE TITLE
Image order not applied correctly in some cases

### ISSUE THREAD

**Issue by manuelmoreale** [getkirby/kirby#2342]
**Describe the bug**  
TLDR: A files section inside site.yml doesn't correctly apply the sort attribute when a `parent`
 attribute is also present.

I have this files section inside site.yml

```
pictures:
  headline : Moments
  type     : files
  status   : all
  template : moment
  layout   : cards
  size     : small
  empty    : No moments... yet
  parent   : site.homePage
  sortable : true
```

The images are correctly displayed and are uploaded to the homepage.
I can sort them in the panel but the `sort` value is not added to the image txt file.
It works as expected if I remove the `parent:` option.

**To Reproduce**  
1. Add the section above inside site.yml
2. Open the panel
3. Upload a couple of images
4. Sort them using manual sorting
5. Reload the panel (or check in the file.txt to see if the sort attribute is there)

**Expected behavior**  
Images stay sorted and sort attribute is added inside the txt file.

**Kirby Version**  
3.3.1

**Comment by distantnative** [comment]
@bastianallgeier I feel like this should be a simple `else`: https://github.com/getkirby/kirby/blob/master/config/sections/files.php#L88

No matter if sortable or not, the files should be sorted in the first place, right?

**Comment by manuelmoreale** [comment]
@distantnative not sure if it's related or not but what looks weird to me is the fact that the sorting value is not added to the file's txt file.

**Comment by trych** [comment]
Had the same bug and I can confirm that there is no sort field added to the images' txt files.

**Comment by manuelmoreale** [comment]
@distantnative news on this one? I just re-open an old project of mine—the one where I discovered the bug in the first place—and noticed the bug is still there in latest Kirby.

**Comment by distantnative** [comment]
Closing this in favor of https://github.com/getkirby/kirby/issues/2690 - which has the more recent discussion.
